internal/controllers/machine/machine_controller_phases_test.go

```diff
@@ -693,131 +692,96 @@ func TestReconcileBootstrap(t *testing.T) {
 				},
 				"spec": map[string]interface{}{},
 				"status": map[string]interface{}{
-					"ready": true,
+					"ready":          false,
+					"dataSecretName": "secret-data",
 				},
 			},
-			expectResult: ctrl.Result{},
-			expectError:  true,
+			bootstrapConfigGetError: nil,
+			expectResult:            ctrl.Result{},
+			expectError:             false,
 			expected: func(g *WithT, m *clusterv1.Machine) {
 				g.Expect(m.Status.BootstrapReady).To(BeFalse())
 				g.Expect(m.Spec.Bootstrap.DataSecretName).To(BeNil())
 			},
 		},
 		{
-			name: "new machine, bootstrap config not ready",
+			name:    "bootstrap config ready with data, it should reconcile and data should surface on the machine",
+			machine: defaultMachine.DeepCopy(),
 			bootstrapConfig: map[string]interface{}{
 				"kind":       "GenericBootstrapConfig",
 				"apiVersion": "bootstrap.cluster.x-k8s.io/v1beta1",
 				"metadata": map[string]interface{}{
 					"name":      "bootstrap-config1",
 					"namespace": metav1.NamespaceDefault,
 				},
-				"spec":   map[string]interface{}{},
-				"status": map[string]interface{}{},
-			},
-			expectResult: ctrl.Result{},
-			expectError:  false,
-			expected: func(g *WithT, m *clusterv1.Machine) {
-				g.Expect(m.Status.BootstrapReady).To(BeFalse())
-			},
-		},
-		{
-			name: "new machine, bootstrap config is not found",
-			bootstrapConfig: map[string]interface{}{
-				"kind":       "GenericBootstrapConfig",
-				"apiVersion": "bootstrap.cluster.x-k8s.io/v1beta1",
-				"metadata": map[string]interface{}{
-					"name":      "bootstrap-config1",
-					"namespace": "wrong-namespace",
+				"spec": map[string]interface{}{},
+				"status": map[string]interface{}{
+					"ready":          true,
+					"dataSecretName": "secret-data",
 				},
-				"spec":   map[string]interface{}{},
-				"status": map[string]interface{}{},
 			},
-			expectResult: ctrl.Result{RequeueAfter: externalReadyWait},
-			expectError:  false,
+			bootstrapConfigGetError: nil,
+			expectResult:            ctrl.Result{},
+			expectError:             false,
 			expected: func(g *WithT, m *clusterv1.Machine) {
-				g.Expect(m.Status.BootstrapReady).To(BeFalse())
-			},
-		},
-		{
-			name: "new machine, no bootstrap config or data",
-			bootstrapConfig: map[string]interface{}{
-				"kind":       "GenericBootstrapConfig",
-				"apiVersion": "bootstrap.cluster.x-k8s.io/v1beta1",
-				"metadata": map[string]interface{}{
-					"name":      "bootstrap-config1",
-					"namespace": "wrong-namespace",
-				},
-				"spec":   map[string]interface{}{},
-				"status": map[string]interface{}{},
+				g.Expect(m.Status.BootstrapReady).To(BeTrue())
+				g.Expect(m.Spec.Bootstrap.DataSecretName).NotTo(BeNil())
+				g.Expect(*m.Spec.Bootstrap.DataSecretName).To(ContainSubstring("secret-data"))
```

Comment:
```suggestion
				g.Expect(*m.Spec.Bootstrap.DataSecretName).To(Equal("secret-data"))
```
I hope (same below in a few more cases)

I think also instead of BeEquivalentTo in l.822